File: Python/greenherb/auth/password_hasher.py
import hashlib


class PasswordHasher:
    def hash(self, password: str) -> str:
        return hashlib.sha256(password.encode("utf-8")).hexdigest()

    def verify(self, password: str, password_hash: str) -> bool:
        return self.hash(password) == password_hash

File: JS/src/services/PasswordHasher.js
import bcrypt from 'bcrypt';

/**
 * Wrapper sobre bcrypt para facilitar mocking em testes.
 */
export class PasswordHasher {
  constructor({ rounds = 10 } = {}) {
    this.rounds = rounds;
  }

  async hash(plainPassword) {
    if (typeof plainPassword !== 'string' || plainPassword.length === 0) {
      throw new Error('Password must be a non-empty string');
    }
    return bcrypt.hash(plainPassword, this.rounds);
  }

  async verify(plainPassword, hash) {
    if (typeof plainPassword !== 'string' || typeof hash !== 'string') {
      return false;
    }
    return bcrypt.compare(plainPassword, hash);
  }
}
